src/v/config/configuration.cc

```diff
@@ -1836,22 +1836,28 @@ configuration::configuration()
   , cloud_storage_max_partition_readers_per_shard(
```

Comment:
I wouldn't overload this config with a new meaning (I find it confusing and I'm not sure what happens on upgrades/rollbacks). I suggest we deprecate and add a new one.